myFslInstallerScript.sh

```diff
@@ -0,0 +1,182 @@
+                 md5Url=$main_url"fsl-5.0.7-centos6_64.tar.gz.md5"
+                 echo "$md5Url"
+         elif [[ "$os" == "centos5" ]]; then
+                 md5Url=$main_url"fsl-5.0.7-centos5_64.tar.gz.md5"
+                 echo "$md5Url"
+         fi
+elif [[ "$version" == "5.0.8" ]]; then
+         if [[ "$os" == "centos7" || "$os" == "centos6" ]]; then
+                 md5Url=$mfain_url"fsl-5.0.8-centos6_64.tar.gz.md5"
+                 echo "$md5Url"
+         elif [[ "$os" == "centos5" ]]; then
+                 md5Url=$main_url"fsl-5.0.8-centos5_64.tar.gz.md5"
+                 echo "$md5Url"
+         fi
+fi
+}
+
+#Parameters: Version and OS
+#Function getDownloadURL will return the url for downloading the requested version.
+function getDownloadURL {
+version="$1"
+os="$2"
+os="${os,,}"
+#echo "$os"
+if [[ ( "$version" == "5.0.6" && "$os" == "centos7" ) || ( "$version" == "5.0.6" && "$os" == "centos6" ) ]]; then
+	echo  'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.6-centos6_64.tar.gz'
+elif [[ ( "$version" == "5.0.6" && "$os" == "centos5" ) ]]; then
+        echo 'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.6-centos5_64.tar.gz'
+elif [[ ( "$version" == "5.0.7" && "$os" == "centos6" ) || ( "$version" == "5.0.7" && "$os" == "centos7" ) ]]; then
+        echo 'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.7-centos6_64.tar.gz'
+elif [[ ( "$version" == "5.0.7" && "$os" == "centos5" ) ]]; then
+        echo 'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.7-centos5_64.tar.gz'
+elif [[ ( "$version" == "5.0.8" && "$os" == "centos6" ) || ( "$version" == "5.0.8" && "$os" == "centos7" ) ]]; then        
+	echo 'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.8-centos6_64.tar.gz'
+elif [[ ( "$version" == "5.0.8" && "$os" == "centos5" ) ]]; then
+        echo 'http://fsl.fmrib.ox.ac.uk/fsldownloads/oldversions/fsl-5.0.8-centos5_64.tar.gz'
+else
+ echo 'Invalid parameters. Check the input parameters and try again'
+ exit 1
+fi
+}
+
+#Parameter:Download URL
+#Function validURL will return true or false according to the status of the url
+function validURL {
+if [[ `wget -S --spider $1  2>&1 | grep 'HTTP/1.1 200 OK'` ]]; then echo "true";fi
+}
+
+# Download function downloads the file in corresponding directory which we pass as the second parameter.
+function download {
+downloadUrl="$1"
+fileName="$2"
+#echo "Downloading.."
+wget $downloadUrl -O $fileName
+}
+
+#validateMD5 function is used for vaidating the md5 checksum value of the downloaded file with the value obtained from the server.
+function validateMD5 {
+echo "Validating MD5 values ..."
+md5FileName="$1"
+fslFileName="$2"
+value=$(awk '{print $1}' "$md5FileName")
+#value=$(<"$md5FileName")
+#echo "MD5"
+#echo "$value"
+localmd5=$(md5sum < "$fslFileName" | awk '{print $1}')
+#echo "$localmd5"
+rm "$md5FileName"
+if [[ "$value" == "$localmd5" ]];then
+	untar "$fslFileName"
```

Comment:
I don't think you're supposed to "untar" the archive in a function called "validateMD5". You should extract the untar call from it.